docs/getting-started/roles/sqlalchemy/basic_roles.rst

```diff
@@ -0,0 +1,270 @@
+=====================
+Roles with SQLAlchemy
+=====================
+
+The :py:data:`sqlalchemy_oso.roles` module provides out-of-the-box Role-Based Access Control features that
+let you create a roles system  with a few lines of code, and specify role permissions
+in a declarative oso policy.
+
+This guide walks you through how to use ``sqlalchemy_oso`` to add basic roles to a multi-tenant app.
+
+    Note: we're using a Flask app for this example, but the
+    ``sqlalchemy_oso`` library can be used with any Python application.
+
+1. Set up the application
+=========================
+
+Install the oso SQLAlchemy package
+----------------------------------
+
+Install the ``sqlalchemy_oso`` package and import it in your code.
+Alternatively, if you are starting from scratch and need a
+``requirements.txt`` file, clone the `sample application
+<https://github.com/osohq/oso-sqlalchemy-roles-guide/tree/main>`_.
+
+.. code-block:: console
+    :class: copybutton
+    :caption: $_
+
+    $ pip install sqlalchemy_oso
+
+
+Add a method to initialize oso and make the oso instance available to your application code.
+Call :py:func:`sqlalchemy_oso.roles.enable_roles` to load the base oso policy for roles.
+
+.. literalinclude:: /examples/roles/sqlalchemy/oso-sqlalchemy-roles-guide/app/__init__.py
+    :class: copybutton
+    :language: python
+    :lines: 11-14, 57-66
+    :caption: :fab:`python` __init__.py
+
+
+
+
+Create a users model
+---------------------
+
+Add a user model that will represent your app's users (if you don't already have one).
+
+
+.. literalinclude:: /examples/roles/sqlalchemy/oso-sqlalchemy-roles-guide/app/models.py
+    :class: copybutton
+    :language: python
+    :lines: 9-10, 14-21
+    :caption: :fab:`python` models.py
+
+
+Create an organizations model
+------------------------------
+
+Add an organization model that will represent the organizations or tenants
+that users belong to. The roles you create will be scoped to this model. If
+your app isn't multi-tenant, you can create an ``Application`` model instead,
+with one instance used globally.
+
+
+.. literalinclude:: /examples/roles/sqlalchemy/oso-sqlalchemy-roles-guide/app/models.py
+    :class: copybutton
+    :language: python
+    :lines: 24-32
+    :caption: :fab:`python` models.py
+
+
+
+Add an endpoint that needs authorization
+----------------------------------------
+
+Create or choose an existing endpoint that will need authorization to access.
+In our sample app we've created 2 endpoints that have different authorization
+requirements: one to view repositories and another to view billing
+information.
+
+Add policy checks to your code to control access to the protected endpoints.
```

Comment:
If you haven't seen oso it might not be clear what `authorize` does, a brief one line description might be helpful.

Reply:
Also at this point with no policy everything will fail, right?

Reply:
Everything will be forbidden yes